### PR TITLE
refactor: export app.HttpClient that can be overwritten

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message follows commit guidelines

### CODE REVIEW

File: lib/core/dnscache_httpclient.js [hunk 1]
```diff
@@ -3,16 +3,14 @@
...
 const utility = require('utility');
 
 const IP_REGEX = /^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$/;
 
-class DNSCacheHttpClient extends urllib.HttpClient {
-  constructor(options) {
-    super(options);
-
-    this.app = options.app;
+class DNSCacheHttpClient extends HttpClient {
```
Comment (popomore) [inline lib/core/dnscache_httpclient.js]: @fengmk2 原来这个类也没继承 core httpclient

File: test/fixtures/apps/httpclient-overwrite/app.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const assert = require('assert');
+
+module.exports = app => {
+  class CustomHttpClient extends app.HttpClient {
```
Comment (popomore) [inline test/fixtures/apps/httpclient-overwrite/app.js]: @leoner 这样继承就可以了

Comment (leoner) [inline test/fixtures/apps/httpclient-overwrite/app.js]: 好

**Comment by popomore** [comment]
可以合了不

**Pull request merged**